compiler/rustc_middle/src/ty/consts.rs

```diff
@@ -26,6 +28,20 @@ use super::sty::ConstKind;
 #[rustc_pass_by_value]
 pub struct Const<'tcx>(pub(super) Interned<'tcx, WithCachedTypeInfo<ConstData<'tcx>>>);
 
+impl<'tcx> IntoKind for Const<'tcx> {
+    type Kind = ConstKind<'tcx>;
+
+    fn kind(&self) -> ConstKind<'tcx> {
+        (*self).kind().clone()
+    }
+}
+
+impl<'tcx> ConstTy<TyCtxt<'tcx>> for Const<'tcx> {
+    fn ty(&self) -> Ty<'tcx> {
```

Comment:
take `self` by value? I don't think we should support consts and types which aren't copy